src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectContextFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal interface IProjectContextFactory
+    {
+        IProjectContext CreateProjectContext(string languageName, string projectDisplayName, string projectFilePath, Guid projectGuid, string projectTypeGuid, object hostObject, CommandLineArguments commandLineArguments);
```

Comment:
We are getting rid of guids from the project file, they'll still be associated with projects and stay in the solution file. I have added the GUID parameter to this method as AbstractProject sets it in the ctor and I didn't want to change those semantics. We can get retain get rid of GUID/projectType guid parameters to this method if we feel CPS won't set them here. I already added methods on IProjectContext to set these explicitly at a later time.